### DISCUSSION TITLE
It only works with Python 3.9.9/3.8-3.11?

### DISCUSSION THREAD

**Discussion by HubKing** [openai/whisper#2348] [Q&A] [answered]
The front page says:

> We used Python 3.9.9 and [PyTorch](https://pytorch.org/) 1.10.1 to train and test our models, but the codebase is expected to be compatible with Python 3.8-3.11 and recent PyTorch versions

Do I have to use the exact version of one of the two? I already have installed Python 3.11 and PyTorch 2.3, so I wonder if I could use that. Arch Linux.

**Comment by glangford** [comment, marked as answer]
I have used Python 3.11.7 with torch 2.2.2 (Mac) successfully.

**Comment by Et0h** [comment]
I have used Python 3.12.6 64-bit and Torch 2.4.1 (Windows 10) successfully.

**Reply by felipeam10** [reply]
It's works too for me using Python 3.12.6 64-bit and Torch 2.4.1 (Windows 10)
I tried using python 3.13/3.10/3.9 and it didn't work.
thank you for help

**Comment by filtercodes** [comment]
Python 3.10.9 here with torch==2.2.1<br/><br/>  -> `RuntimeError: Given groups=1, weight of size [1280, 128, 3], expected input[1, 80, 3000] to have 128 channels, but got 80 channels instead`<br/><br/> Mac OS metal though

**Reply by glangford** [reply]
@filtercodes This is likely because an older version of whisper is being used in combination with the large-v3 model. The input to large-v3 uses 128 Mel frequency bins instead of 80. 

From the announcement,

> The large-v3 model is available in openai-whisper==20231106 and after

* https://github.com/openai/whisper/discussions/1762

**Reply by filtercodes** [reply]
> @filtercodes This is likely because an older version of whisper is being used in combination with the large-v3 model. The input to large-v3 uses 128 Mel frequency bins instead of 80.
> 
> From the announcement,
> 
> > The large-v3 model is available in openai-whisper==20231106 and after
> 
> * [`large-v3` release #1762](https://github.com/openai/whisper/discussions/1762)

This is the version that shows up ->  `openai-whisper @ git+https://github.com/openai/whisper.git@25639fc17ddc013d56c594bfbf7644f2185fad84`  <br/><br/>Command I was running was: `pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git` and that was supposed to update whisper to the latest version. Not sure what I'm missing here....

**Reply by glangford** [reply]
Is the code calling `log_mel_spectrogram` directly? If so you can specify the number of channels  with `n_mels`.

* https://github.com/openai/whisper/discussions/1778#discussioncomment-7520324

To confirm that the right version of whisper is being called (and the Python code is not an issue), you could try using the command line with large-v3.

**Reply by filtercodes** [reply]
> Is the code calling `log_mel_spectrogram` directly? If so you can specify the number of channels with `n_mels`.
> 
> * [Error in the "large-v3" model #1778 (comment)](https://github.com/openai/whisper/discussions/1778#discussioncomment-7520324)
> 
> To confirm that the right version of whisper is being called (and the Python code is not an issue), you could try using the command line with large-v3.

That was it :) n_mels has to be redefined because it uses 80 as a default even it gets it from model.dims.n_mels. <br/>Also it works only on CPU, not on metal. A simple to("mps") didn't do the trick. Then a (mostly) single threaded inference takes couple of minutes, which doesn't sound very "turbo" at all

**Reply by glangford** [reply]
Good news. :)

mps acceleration is still a work in progress, for "too much history" see the thread below. For the record, the turbo model on CUDA is insanely quick.

* https://github.com/pytorch/pytorch/issues/77764
